core/src/main/java/org/elasticsearch/action/admin/cluster/allocation/ClusterAllocationExplainRequest.java

```diff
@@ -245,6 +247,7 @@ public void readFrom(StreamInput in) throws IOException {
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
+        checkVersion(out.getVersion());
```

Comment:
I wonder if we can have a REST test that checks this exception. This could be done with a skip section of the form `5.2.0 - ` so that the test is only active in BWC tests where the minimum es version is < 5.2.0.